gax-httpjson/src/main/java/com/google/api/gax/httpjson/ApiMessage.java

```diff
@@ -30,16 +30,13 @@
 package com.google.api.gax.httpjson;
 
 import com.google.api.core.BetaApi;
-import java.util.List;
-import java.util.Map;
-import java.util.Set;
 import javax.annotation.Nullable;
 
 @BetaApi
-public interface ApiMessage {
-  Map<String, List<String>> populateFieldsInMap(Set<String> fieldNames);
+public interface ApiMessage extends Struct {
+  ResourceNamePath resourceNamePath();
 
   @Nullable
   /* If this is a Request object, return the inner ApiMessage that represents the body of the request; else return null. */
-  ApiMessage getRequestBody();
```

Comment:
We use the `getX()` pattern throughout our code - what is the reason for changing it here?

Reply:
In case there's a naming conflict where one of the message fields is called "requestBody", this overridden interface method has a different name.

Reply:
That sounds horrible... maybe we should disallow that?